dashboard/modules/serve/sdk.py

```diff
@@ -30,6 +31,24 @@ def __init__(
                 "The Serve CLI requires the ray[default] "
                 "installation: `pip install 'ray[default']``"
             )
+
+        module_string, _ = _split_address(
+            dashboard_agent_address, prepend_ray_if_needed=False
+        )
+
+        # If user passes in ray://, raise error. Serve submission should
+        # not use a Ray client address.
+        if module_string == "ray":
```

Comment:
i just pass "x.x.x.x" without module_string as `dashboard_agent_address`, what error I will get?

Reply:
I might prefer to have the address check rule inside each class and let the `_split_address` do the general job and even remove the 
```
    if "://" not in address:
        address = "ray://" + address
```

BTW, why does the `_split_address` is a private function but it is being used outside? (maybe some legacy stuff)

Reply:
Good point, it will go into the constructor for the base class and then probably fail when it tries to make any request.  Should we just enforce `module_string in {"http", "https://"}` here? cc @shrekris-anyscale

Reply:
Not sure if there's a historical reason `_split_function` is being used, it should be safe for me to move it into a `util` file and make it public.  Let me look into this